### DISCUSSION TITLE
Force vehicle immediately going back to depot only after last location

### DISCUSSION THREAD

**Discussion by kazitanvirahsan** [google/or-tools#2799] [Routing (and legacy CP) questions]
Or gurus,

   I need some help so that I can force the vehicle to return to the depot immediately from the last location. At the moment, vehicles keep waiting to the last location as much as they can before returning to the depot.

  Here is my time dimention,
 
  time = 'Time'
    routing.AddDimension(
        time_callback_index,
        30000,  # allow waiting time
        30000,  # maximum time per vehicle
        False,  # Don't force start cumul to zero.
        time)
    time_dimension = routing.GetDimensionOrDie(time)

   I allow high waiting time to each node so that it can wait in the location to pick the next load of opening hours.
However, I really want that Once all the pickups and drops off are done, the vehicle does not need to wait in the last location and immediately going back to the depot.
   

<img width="1021" alt="Screen Shot 2021-09-30 at 11 26 39 am" src="https://user-images.githubusercontent.com/6487015/135370734-bc1c4180-853d-44c7-b5e0-f8185e3f5a4e.png">

**Comment by Mizux** [comment]
question: do you have a TW on the vehicle end node ? seems strange that the vehicle prefer to wait than returning ASAP
Also, you may try to use a finalizer on the vehicle end node
https://github.com/google/or-tools/blob/b37d9c786b69128f3505f15beca09e89bf078a89/ortools/constraint_solver/samples/vrp_time_windows.py#L164-L165

**Comment by paultrow** [comment]
If there is slack time at any node, the driver can do whatever they want
during that time before proceeding to the next node. The same is true at
the last node. Telling drivers what to do during slack time isn't really
part of the VRPTW problem.
